lib/libc/string/wcscasecmp.c

```diff
@@ -35,11 +35,13 @@ wcscasecmp(const wchar_t *s1, const wchar_t *s2)
 {
 	wchar_t c1, c2;
 
-	for (; *s1; s1++, s2++) {
+	for (;; s1++, s2++) {
 		c1 = towlower(*s1);
 		c2 = towlower(*s2);
 		if (c1 != c2)
-			return ((int)c1 - c2);
+			return ((unsigned int)c1 > (unsigned int)c2 ? 1 : -1);
```

Comment:
I fail to see the benefit from this cast. What's the reason for it?

Reply:
Because wide char types are assumed to be 32 bit. This casting wouldn’t be needed if these were regular chars